platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractProperty.java

```diff
@@ -28,6 +28,20 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+/**
+ * The base implementation for all properties in Gradle.
```

Comment:
❓ There is automation that enforces this, correct?  We should note that here, too.

Reply:
Now there is ;) 

Do we really need to add a note here? It is implied in the summary for this class, right? Also, I don't think anyone looking into adding a new property would miss this base class. But if they somehow did, the new architectural rule will remind them.